### PR TITLE
feat(babel): use loose mode

### LINKED ISSUE #32: Loose mode by default
I suggest using babel's `loose` mode by default, it allows you to ship quite less bytes of code to the end users and truth to be told, you shouldn't really miss most of default fully spec-compliant behaviours.

### PR DESCRIPTION
**Pull request by Andarist**
fixes #32

**What**:

`loose` mode for babel transforms

**Why**:

shorter, more optimized output code

**Checklist**:

* [ ] Documentation
* [ ] Tests
* [x] Ready to be merged 
* [ ] Added myself to contributors table

### CODE REVIEW

File: src/config/babelrc.js [hunk 1]
```diff
@@ -42,8 +41,12 @@ module.exports = {
     isUMD
       ? require.resolve('babel-plugin-transform-inline-environment-variables')
       : null,
+    // TODO: use loose mode when upgrading to babel@7
     require.resolve('babel-plugin-transform-class-properties'),
     require.resolve('babel-plugin-transform-object-rest-spread'),
     require.resolve('babel-plugin-minify-dead-code-elimination'),
+    !treeshake
+      ? require.resolve('babel-plugin-transform-es2015-modules-commonjs')
```
Comment (Andarist) [inline src/config/babelrc.js]: modules-commonjs also has a `loose` mode, but i dont think its worth using as it breaks namespace imports

```js
import * as namespace from './namespace'
```

in `loose` namespace would have **enumerable** `__esModule` property which can surprise quite a bit when spreading a namespace

File: src/config/babelrc.js [hunk 2]
```diff
@@ -8,11 +8,10 @@ const isWebpack = parseEnv('BUILD_WEBPACK', false)
 const treeshake = parseEnv('BUILD_TREESHAKE', isRollup || isWebpack)
 const alias = parseEnv('BUILD_ALIAS', isPreact ? {react: 'preact'} : null)
 
-const envModules = treeshake ? {modules: false} : {}
 const envTargets = isTest
   ? {node: 'current'}
   : isWebpack || isRollup ? {browsers: ['ie 10', 'ios 7']} : {node: '4.5'}
-const envOptions = Object.assign({}, envModules, {targets: envTargets})
+const envOptions = {modules: false, loose: true, targets: envTargets}
```
Comment (kentcdodds) [inline src/config/babelrc.js]: Hmmm... We only want `{modules: false}` if `treeshake` is true right?

Comment (Andarist) [reply src/config/babelrc.js]: im adding transform-es2015-commonjs "by hand" few line below, see [this comment](https://github.com/kentcdodds/kcd-scripts/pull/33#discussion_r166554943)

Comment (kentcdodds) [reply src/config/babelrc.js]: Ah, sorry, I somehow missed that. Looks good!

**Pull request merged**